model.py

```diff
@@ -8654,10 +8721,23 @@ class Collection(Base):
     )
 
     @property
+    def external_integration(self):
+        _db = Session.object_session(self)
+        if self.external_integration_id:
+            external_integration = get_one(
+                _db, ExternalIntegration, id=self.external_integration_id,
+            )
+        else:
+            external_integration, ignore = create(
+              _db, ExternalIntegration
+            )
+            self.external_integration_id = external_integration.id
+        return external_integration
```

Comment:
Ditto here -- I get that this is a little different from a normal scenario where the foreign key is managed with `relationship` but `get_one_or_create` seems like it would be safer. Is there a reason why it won't work here?